curriculum/challenges/english/25-front-end-development/review-loops-and-sequences/67f39b91583dec1e1eddbb9e.md

```diff
@@ -7,13 +7,703 @@ dashedName: review-loops-and-sequences
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust')
+programming_languages.count('JavaScript') # 0
+```
+
+- If no arguments are passed to the `count()` function, then Python will return a `TypeError`.
+
+- **index()**: Used to find the index where a particular item is present in the tuple.  Here is an example of using the `index()` method to find the index for the language `'Java'`:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust')
+programming_languages.index('Java') # 1
+```
+
+- If the specified item cannot be found, then Python will return a `ValueError`.
+
+- You can pass an optional start index to the `index()` method to specify where to start searching for the item in the tuple:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust', 'Python')
+programming_languages.index('Python', 3) # 5
+```
+
+- You can also pass in an optional end index to the `index()` method to specify where to stop searching for the item in the tuple:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust', 'Python', 'JavaScript', 'Python')
+programming_languages.index('Python', 2, 5) # 2
+```
+
+- **`sorted()`**: Used to sort the elements in any iterable and return a new sorted list.  Here is an example of creating a new list of numbers using the `sorted()` function:
+
+```py
+numbers = (13, 2, 78, 3, 45, 67, 18, 7)
+sorted(numbers) # [2, 3, 7, 13, 18, 45, 67, 78]
+```
+
+- **Modifying Sorting Behavior**: You can customize the sorting behavior for an iterable using the optional `reverse` and `key` arguments.  Here is an example of using the `key` argument to sort items in a tuple by length:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust', 'Python')
+sorted(programming_languages, key=len)
+
+# Result
+# ['C++', 'Rust', 'Java', 'Rust', 'Python', 'Python']
+```
+
+- You can create a new list of values in reverse order, using the `reverse` argument like this:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust', 'Python')
+
+print(sorted(programming_languages, reverse=True))
+
+# Result
+# ['Rust', 'Rust', 'Python', 'Python', 'Java', 'C++']
+```
+
+## Loops in Python
+
+- **Definition**: Loops are used to repeat a block of code for a set number of times.
+
+- **`for` loop**: Used to iterate over a sequence (like a list, tuple or string) and execute a block of code for each item in that sequence.  Here is an example of using a `for` loop to iterate through a list and print each language to the console:
+
+```py
+programming_languages = ['Rust', 'Java', 'Python', 'C++']
+
+for language in programming_languages:
+    print(language)
+
+"""
+Result 
+
+'Rust'
+'Java'
+'Python'
+'C++'
+"""'''
+```
+
+- Here is an example of using a `for` loop to loop through the string `'code'` and print out each character:
+
+```py
+for char in 'code':
+    print(char)
+
+"""
+Result 
+
+'c'
+'o'
+'d'
+'e'
```

Comment:
```suggestion
c
o
d
e
```
I saw this many times, (also in other files) but we should stop showing quotes around strings where they are not actually displayed by printing stuff.